src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -155,6 +155,15 @@ private static string NormalizeStringCollectionItems(string literalText)
 
             // remove the '[' and ']'
             string normalizedText = literalText.Substring(1, literalText.Length - 2).Trim();
+
+            string normalizedTextWithoutWhitespaces = normalizedText.Replace(" ", string.Empty);
+
+            // If we have '' or "" or empty string we return empty brackets.
+            if (normalizedTextWithoutWhitespaces == "''" || normalizedTextWithoutWhitespaces == "\"\""  || normalizedTextWithoutWhitespaces.Length == 0)
+            {
+                return "[]";
+            }

```

Comment:
Why do we return empty brackets here? Isn't an empty string a valid string value? I assume a collection with a single empty string value is still a non-empty collection. Why are we removing the value from the collection?

Reply:
I agree with Clement that  and empty string is a string and 
a collection with only an empty string has the same number of elements as a collection with the string "a"

Reply:
I also agree. Unfortunately with can't create a ConstantNode with an empty string. However we can create a ConstantNode with single/multiple whitespaces.
https://github.com/OData/odata.net/blob/6003304b3bbb58c91d1bd6f6ac9f43d1e50deb82/src/Microsoft.OData.Core/UriParser/SemanticAst/ConstantNode.cs#L53

Reply:
This is broken.  It should be valid to have an empty string in the collection.

What would break if we made ConstantNode support string.Empty?

name in ('') should be equivalent to name eq '' -- how do we parse something like name eq '' if we can't create an empty string in a constant expression?

Reply:
To be very clear, the constructor `ConstantNode(object constantValue, string literalText)` ensures that literalText is not null or empty.
But `new ConstantNode("", "\"\"")` should be valid and seems to be implemented that way.